pulpcore/cli/python/content.py

```diff
@@ -0,0 +1,175 @@
+import gettext
+from typing import IO, Any, Optional, Union
+
+import click
+
+from pulpcore.cli.common.context import (
+    PulpContext,
+    PulpEntityContext,
+    pass_entity_context,
+    pass_pulp_context,
+)
+from pulpcore.cli.common.generic import create_command, href_option, list_command, show_command
+from pulpcore.cli.core.context import PulpArtifactContext
+from pulpcore.cli.python.context import PulpPythonContentContext
+
+_ = gettext.gettext
+
+
+def _sha256_artifact_callback(
+    ctx: click.Context, param: click.Parameter, value: Optional[str]
+) -> Optional[Union[str, PulpEntityContext]]:
+    # Pass None and "" verbatim
+    if value:
+        pulp_ctx: PulpContext = ctx.find_object(PulpContext)
+        return PulpArtifactContext(pulp_ctx, entity={"sha256": value})
+    return value
+
+
+@click.group()
+@click.option(
+    "-t",
+    "--type",
+    "content_type",
+    type=click.Choice(["package"], case_sensitive=False),
+    default="package",
+)
+@pass_pulp_context
+@click.pass_context
+def content(ctx: click.Context, pulp_ctx: PulpContext, content_type: str) -> None:
+    if content_type == "package":
+        ctx.obj = PulpPythonContentContext(pulp_ctx)
+    else:
+        raise NotImplementedError()
+
+
+create_options = [
+    click.option("--relative-path", required=True),
+    click.option(
+        "--sha256",
+        "artifact",
+        required=True,
+        help=_("Digest of the artifact to use"),
+        callback=_sha256_artifact_callback,
+    ),
+]
+python_content_options = [
+    click.option("--summary", help=_("A one-line summary of what the package does.")),
+    click.option(
+        "--description",
+        help=_("A longer description of the package that can run several paragraphs."),
+    ),
+    click.option(
+        "--keywords",
+        help=_(
+            "Additional keywords to be used to assist searching for the package in a larger"
+            " catalog."
+        ),
+    ),
+    click.option("--home_page", help=_("The URL for the package's home page.")),
+    click.option(
+        "--download_url",
+        help=_("Legacy field denoting the URL from which this package can be downloaded."),
+    ),
+    click.option(
+        "--author",
+        help=_(
+            "Text containing the author's name. Contact information can also be added, separated"
+            " with newlines."
+        ),
+    ),
+    click.option("--author_email", help=_("The author's e-mail address.")),
+    click.option(
+        "--maintainer",
+        help=_(
+            "The maintainer's name at a minimum; additional contact information may be provided."
+        ),
+    ),
+    click.option("--maintainer_email", help=_("The maintainer's e-mail address.")),
+    click.option("--license", help=_("Text indicating the license covering the distribution")),
+    click.option(
+        "--requires_python",
```

Comment:
```suggestion
        "--requires-python",
```
This holds true for the other options as well.